### PR TITLE
tracing: improved db logging

### PR DESCRIPTION
**Pull request by Maan2003**
Now you can view all changes to database.

### CODE REVIEW

File: fedimint-api/src/db/mod.rs [hunk 1]
```diff
@@ -25,8 +25,8 @@ pub const MODULE_GLOBAL_PREFIX: u8 = 0xff;
 
 pub trait DatabaseKeyPrefixConst {
     const DB_PREFIX: u8;
-    type Key: DatabaseKey;
-    type Value: DatabaseValue;
+    type Key: DatabaseKey + Debug;
+    type Value: DatabaseValue + Debug;
```
Comment (Maan2003) [inline fedimint-api/src/db/mod.rs:29]: A small price to pay for debugging.
should be fine IMO

**Comment by Maan2003** [comment]
also adding `docs/debugging.md`, like @justinmoon suggested.

File: docs/debugging.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Debugging Helpers
+
+- To log all database requests for a specific key: `RUST_LOG="fedimint_api[{key=ContractKey.*}]=debug"`.
```
Comment (dpc) [inline docs/debugging.md:3]: I do not immediately understand this syntax. Please give an example and maybe link to a doc of the crate that describes how it works.

**Comment by dpc** [comment]
@m1sterc001guy

File: fedimint-api/src/db/mod.rs [hunk 3]
```diff
@@ -357,6 +357,18 @@ impl<'a> std::ops::DerefMut for DatabaseTransaction<'a> {
     }
 }
 
+#[instrument(level = "trace", skip_all, fields(value_type = std::any::type_name::<V>()), err)]
```
Comment (m1sterc001guy) [inline fedimint-api/src/db/mod.rs:360]: Just for my understanding, what does this do?

File: fedimint-api/src/db/mod.rs [hunk 4]
```diff
@@ -407,21 +420,21 @@ impl<'parent> DatabaseTransaction<'parent> {
...
 
+    #[instrument(level = "debug", skip_all, fields(key = ?key_prefix))]
     pub async fn find_by_prefix<KP>(
         &mut self,
         key_prefix: &KP,
     ) -> impl Iterator<Item = Result<(KP::Key, KP::Value)>> + '_
     where
         KP: DatabaseKeyPrefix + DatabaseKeyPrefixConst,
     {
+        debug!("find by prefix");
```
Comment (m1sterc001guy) [inline fedimint-api/src/db/mod.rs:437]: Is this necessary? Maybe include which prefix?

**Review by m1sterc001guy — APPROVED** [review]

**Comment by Maan2003** [comment]
please rerun the CI, this is an intermittant failure.

**Comment by Maan2003** [comment]
Three intermittant CI failures in a row. Wow

**Review by dpc — APPROVED** [review]

**Review by m1sterc001guy — APPROVED** [review]

**Pull request merged**